Implement IDENT server.

sayanchowdhury opened the pull request.

sayanchowdhury committed:
Initial work on the ident server

rtnpro committed:
Implement working identd server.

slick666 commented:
Don't we want to squash these commits?

slick666 commented on ircb/identd.py:
```diff
@@ -0,0 +1,75 @@
+import asyncio
+import logging
+import logging.config
+import traceback
```
This import isn't used

sayanchowdhury replied:
Thanks, will remove this.

slick666 commented on ircb/identd.py:
```diff
@@ -0,0 +1,75 @@
+
+    @asyncio.coroutine
+    def handle_data_received(self, data):
+        print(data)
```
instead of print shouldn't we use the logging function?

sayanchowdhury replied:
Yes, these print statements will be removed as I will be doing some minor changes the way we log.

slick666 commented on ircb/identd.py:
```diff
@@ -0,0 +1,75 @@
+    @asyncio.coroutine
+    def handle_data_received(self, data):
+        print(data)
+        no_user_tmpl = "{lport}, {rport} : ERROR : NO-USER"
```
These constants should be capitalized

sayanchowdhury replied:
Okay! I will capitalize them.

sayanchowdhury commented:
@slick666 This is a PR in progress, I will update the description. I will be pushing a few more updates like implementing a command to run the server and yes, I would be squashing the commits too.

slick666 commented:
sorry @sayanchowdhury, would you please add the work-in-progress label?

sayanchowdhury commented:
Added :)

rtnpro committed:
Refactor run command to open room for identd server.

- Remove server command from 'run'
- Added 'allinone' and 'bouncer' command to 'run' as first class entities

rtnpro committed:
Refactor allinone CLI command to move implementation outside of bouncer.

rtnpro committed:
Integrate identd server with CLI

- in 'ircb run allinone'
- in 'ircb run identd'

rtnpro committed:
Fix implementation of identd server

Handle received data and send reply in response.

rtnpro committed:
Bugfix during saving connection info for a IRC connection.

sayanchowdhury commented:
:+1: for me
